### DISCUSSION TITLE
Using Argo to rollout ConfigMaps/Secrets/CRDs

### DISCUSSION THREAD

**Discussion by Dentrax** [argoproj/argo-rollouts#2107] [Q&A]
I want to use Argo Rollouts to gradually rollout my ConfigMap/Secret/CRD resources to clusters. Can I use it for this purpose? Or it's just designed to work with Deployments/Services? (If so, are there any workaround or related works to make it work to rollout my custom resources to clusters?)

Use Case:
I want to change my service's configuration gradually across clusters and want to see if everything is OK. I also want to use AnalysisTemplate to check errors from Prometheus metrics.

Thanks!

**Comment by kostis-codefresh** [comment]
Hello

This has already been discussed. 
See https://github.com/argoproj/argo-rollouts/discussions/1930 and https://github.com/argoproj/argo-rollouts/discussions/1346

**Comment by kostis-codefresh** [comment]
@Dentrax Check this guide https://codefresh.io/blog/progressive-delivery-for-kubernetes-config-maps-using-argo-rollouts/

**Reply by afagund** [reply]
Does not work with Helm

**Reply by kostis-codefresh** [reply]
You can do the same thing with Helm and the checksum method https://helm.sh/el/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments

**Reply by afagund** [reply]
That's true. I found that article later and implemented successfully. Thanks for pointing that out!
